src/contract_wrappers/modules/wallet/vesting_escrow_wallet_wrapper/3.1.0.ts

```diff
@@ -0,0 +1,125 @@
+import { VestingEscrowWalletContract_3_1_0, Web3Wrapper, BigNumber, PolyResponse } from '@polymathnetwork/abi-wrappers';
+import assert from '../../../../utils/assert';
+import ContractFactory from '../../../../factories/contractFactory';
+import { ErrorCode, ContractVersion, TxParams, Perm } from '../../../../types';
+import { weiToValue, stringToBytes32, dateToBigNumber } from '../../../../utils/convert';
+import VestingEscrowWalletCommon from './common';
+
+/**
+ * @param beneficiary Address of the beneficiary for whom it is modified
+ * @param templateName Name of the template was used for schedule creation
+ * @param startTime Start time of the created vesting schedule
+ */
+interface ModifyScheduleParams extends TxParams {
+  beneficiary: string;
+  templateName: string;
+  startTime: Date;
+}
+
+// RETURN TYPES
+
+interface Template {
+  numberOfTokens: BigNumber;
+  duration: number;
+  frequency: number;
+  index: number;
+}
+
+/**
+ * This class includes the functionality related to interacting with the Vesting Escrow Wallet contract.
+ */
+export class VestingEscrowWallet_3_1_0 extends VestingEscrowWalletCommon {
+  public contract: Promise<VestingEscrowWalletContract_3_1_0>;
+
+  public contractVersion = ContractVersion.V3_1_0;
+
+  /**
+   * Instantiate VestingEscrowWalletWrapper
+   * @param web3Wrapper Web3Wrapper instance to use
+   * @param contract
+   * @param contractFactory
+   */
+  public constructor(
+    web3Wrapper: Web3Wrapper,
+    contract: Promise<VestingEscrowWalletContract_3_1_0>,
+    contractFactory: ContractFactory,
+  ) {
+    super(web3Wrapper, contract, contractFactory);
+    this.contract = contract;
+  }
+
+  /**
+   * Modifies a vesting schedule for a beneficiary address
+   */
+  public modifySchedule = async (params: ModifyScheduleParams): Promise<PolyResponse> => {
+    assert.assert(
+      await this.isCallerAllowed(params.txData, Perm.Admin),
+      ErrorCode.Unauthorized,
+      'Caller is not allowed',
+    );
+    this.checkSchedule(params.beneficiary);
+    assert.assert(params.startTime.getTime() >= Date.now(), ErrorCode.TooEarly, 'Date in the past');
+    // TODO: require(now < schedule.startTime, "Schedule started");
```

Comment:
Why is this TODO here?

Reply:
referent comment that I forgot to remove